### PR TITLE
Fix defs of "fast_length" and "fast_normalize"

### PR DESCRIPTION
**Pull request by gmlueck**
It appears that these function definitions were incorrectly transcribed from OpenCL.  In OpenCL, they rely on the half-precision versions of `sqrt` and `rsqrt`, not the overloads that take `half` types.

Closes internal issue 656.

### CODE REVIEW

**Comment by gmlueck** [comment]
IMPORTANT!

This is currently a "stacked" PR on top of #440.  Once #440 is merged, this PR should be retargeted to the "SYCL-2020/master" branch.

File: adoc/chapters/programming_interface.adoc [hunk 1]
```diff
@@ -25619,7 +25619,7 @@ _Constraints:_ Available only if [code]#GeoFloat# is a _float geometric type_
 as defined above.
 
 _Returns:_ The length of vector [code]#p# computed as:
-[code]#+sqrt((half)(pow(p[0],2) + pow(p[1],2) + ...))+#.
+[code]#+half_precision::sqrt((pow(p[0],2) + pow(p[1],2) + ...))+#.
```
Comment (keryell) [inline adoc/chapters/programming_interface.adoc:25622]:
```suggestion
[code]#+half_precision::sqrt(pow(p[0],2) + pow(p[1],2) + ...)+#.
```

File: adoc/chapters/programming_interface.adoc [hunk 2]
```diff
@@ -25639,7 +25639,8 @@ _Constraints:_ Available only if [code]#GeoFloat# is a _float geometric type_
 as defined above.
 
 _Returns:_ A vector in the same direction as [code]#p# but with a length of 1
-computed as [code]#+p*rsqrt((half)(pow(p[0],2) + pow(p[1],2) + ... ))+#.
+computed as
+[code]#+p * half_precision::rsqrt((pow(p[0],2) + pow(p[1],2) + ... ))+#.
```
Comment (keryell) [inline adoc/chapters/programming_interface.adoc:25643]:
```suggestion
[code]#+p * half_precision::rsqrt(pow(p[0],2) + pow(p[1],2) + ... )+#.
```

**Review by AerialMantis — APPROVED** [review]

**Review by keryell — APPROVED** [review]
Thanks!

**Comment by tomdeakin** [comment]
Ready to merge.

**Pull request merged**